flamedisx/inference.py

```diff
@@ -391,12 +396,73 @@ def parse_result(self, result: Minuit):
         position = {k: result.fitarg[k] for k in self.arg_names}
         return position, result.fval
 
+class NonlinearObjective(Objective):
+    """Compute limits on target parameter given nonlinear constraint
+    defined by likelihood ratio.
+    """
+
+    require_complete_guess = False
+
+    # TODO: Fix callback
+    # Callback function to capture intermediate states of optimizer
+    #def fishPath(self, a, b):
+    #    self._callbackbag.append(a)
+
+    def _inner_fun_and_grad(self, params):
+        # lol who exactly is this Objective._inner_fun_and_grad(self, params)?
+        m2ll, grad, hess = Objective._inner_fun_and_grad(self, params)
+        return m2ll - self.m2ll_best - self.t_ppf(params), grad - self.t_ppf_grad(params), hess - self.t_ppf_hess(params)
+
+    def tilt_fun(self, x):
+        # so that we get a scalar back regardless of what nuisance parameters we
+        # might have
+        return -self.direction*x[0]
```

Comment:
The current value of `self.target_parameter` may not always be the first element of `x`. So we probably need something like `self._array_to_dict(x)[self.target_parameter]` instead of `x[0]`

Reply:
Pueh solved this in 406d289d169bd2736f46b3618ce04fe93ef38a27